src/components/Environment/EnvironmentCard.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import {
+  CardHeader,
+  Card,
+  CardTitle,
+  CardActions,
+  CardBody,
+  DropdownItem,
+  DropdownSeparator,
+  Dropdown,
+  KebabToggle,
+  Text,
+  TextContent,
+  TextVariants,
+  Badge,
+} from '@patternfly/react-core';
+import { getRunStatusColor, runStatus } from '../../shared';
+import ExternalLink from '../../shared/components/links/ExternalLink';
+import { StatusIcon } from '../../shared/components/pipeline-run-logs/StatusIcon';
+import { EnvironmentKind } from '../../types';
+
+type EnvironmentCardProps = {
+  environment: EnvironmentKind;
+};
+const EnvironmentCard: React.FC<EnvironmentCardProps> = ({ environment }) => {
+  const [isKebabOpen, setKebabOpen] = React.useState<boolean>(false);
+  const dropdownItems = [
+    <DropdownItem key="link">View</DropdownItem>,
+    <DropdownItem key="disabled action" isDisabled component="button">
+      Edit
+    </DropdownItem>,
+    <DropdownSeparator key="separator" />,
+    <DropdownItem key="separated link">Delete</DropdownItem>,
+  ];
+
+  const onToggle = () => {
+    setKebabOpen(!isKebabOpen);
+  };
+
+  const dummyLink = 'www.example.com/cluster-link';
+  const apiServerURL = environment.spec?.clusterCredentials?.apiServerURL;
+
+  return (
+    <Card>
+      <CardHeader>
+        <CardTitle>
+          <TextContent>
+            <Text component={TextVariants.h4}>
+              {environment.spec?.displayName ?? environment.metadata.name}
+            </Text>
+          </TextContent>
+        </CardTitle>
+        <CardActions>
+          <Dropdown
+            toggle={<KebabToggle onToggle={onToggle} />}
+            isOpen={isKebabOpen}
+            isPlain
+            dropdownItems={dropdownItems}
+          />
+        </CardActions>
+      </CardHeader>
+      {(apiServerURL || dummyLink) && (
+        <CardBody>
+          <TextContent>
+            <Text component={TextVariants.h6}>Environment Link:</Text>
+          </TextContent>
+          <ExternalLink
+            href={environment.spec?.clusterCredentials?.apiServerURL || dummyLink}
+            text={environment.spec?.clusterCredentials?.apiServerURL || dummyLink}
+          />
+        </CardBody>
+      )}
+      <CardBody>
+        <TextContent>
+          <Text component={TextVariants.h6}>Deployment Strategy:</Text>
+        </TextContent>
+        <Badge isRead>{environment.spec.deploymentStrategy}</Badge>
+      </CardBody>
+      <CardBody>
+        <TextContent>
+          <Text component={TextVariants.h6}>Environment Status:</Text>
+          <span style={{ color: getRunStatusColor(runStatus.Succeeded).pftoken.value }}>
+            <StatusIcon status={runStatus.Succeeded} />
+          </span>{' '}
```

Comment:
These are pipeline run status  and EnvironmentStatus might be different from these better to use something else for now

Reply:
Omitting the status section until we have clarity on the object structure
https://coreos.slack.com/archives/C02CTEB3MMF/p1657017875592869